Sources/PriorityQueueModule/PriorityQueue.swift

```diff
@@ -173,23 +173,33 @@ public struct PriorityQueue<Element: Comparable> {
     }
   }
 
+  @inline(__always)
   @inlinable
   internal mutating func _bubbleUpMin(startingAt index: Int) {
-    guard let grandparentIdx = _grandparentIndex(of: index) else { return }
-
-    if storage[index] < storage[grandparentIdx] {
-      _swapAt(index, grandparentIdx)
-      _bubbleUpMin(startingAt: grandparentIdx)
+    var index = index
+      
+    while true {
+      guard let grandparentIdx = _grandparentIndex(of: index) else { return }
+
+      if storage[index] < storage[grandparentIdx] {
+          _swapAt(index, grandparentIdx)
+          index = grandparentIdx
+      } else { return }
     }
   }
 
+  @inline(__always)
   @inlinable
   internal mutating func _bubbleUpMax(startingAt index: Int) {
-    guard let grandparentIdx = _grandparentIndex(of: index) else { return }
-
-    if storage[index] > storage[grandparentIdx] {
-      _swapAt(index, grandparentIdx)
-      _bubbleUpMax(startingAt: grandparentIdx)
+    var index = index
+      
+    while true {
+      guard let grandparentIdx = _grandparentIndex(of: index) else { return }
+
+      if storage[index] > storage[grandparentIdx] {
+        _swapAt(index, grandparentIdx)
+        index = grandparentIdx
+      } else { return }
```

Comment:
Same note here about consolidating this to the `while` condition.

Reply:
Done.